src/auth/dto/auth.dto.ts

```diff
@@ -43,3 +43,18 @@ export class SetNewPasswordDto {
   })
   passwordConfirmation: string;
 }
+
+/**
+ * Note: This DTO is only used for better typing for Swagger, since the
+ * passport-local strategy is hardcoded to use email and password seperately
+ * (see jwt.strategy.ts).
+ */
+export class LoginDto {
```

Comment:
I'm not really happy with adding a class only for Swagger. Isn't there another way that represents the current implementation the way it is?

Reply:
No, similar issue as with the file DTO. This is *actually* the DTO that should be used for typing the API because it specifies what a user sends.

I added this comment only because we cannot use the `LoginDto` inside our strategies, because the passport strategy is hardcoded to use these fields.

The only solution I see would be to "copy" the passport strategy code and "implement" it ourselves, changing it to the DTO. But I am not a fan of touching security-related code, because we'd also lose any updates.